src/main/java/hudson/plugins/warnings/WarningsPublisher.java

```diff
@@ -313,15 +329,15 @@ public Action getProjectAction(final AbstractProject<?, ?> project) {
     }
 
     @Override
-    protected BuildResult perform(final AbstractBuild<?, ?> build, final PluginLogger logger)
+    protected BuildResult perform(final Run<?, ?> build, final FilePath workspace, final PluginLogger logger)
```

Comment:
I guess you did not use `@deprecated` because it is `protected` and you can safely modify the original function, correct?